core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -1219,6 +1287,73 @@ public void importTable(String namespace, String table) {
         "import-related functionality is not supported in DynamoDB");
   }
 
+  @Override
+  public Set<String> getNamespaceNames() throws ExecutionException {
+    try {
+      Set<String> namespaceNames = new HashSet<>();
+      Map<String, AttributeValue> lastEvaluatedKey = null;
+      do {
+        ScanResponse scanResponse =
+            client.scan(
+                ScanRequest.builder()
+                    .tableName(ScalarDbUtils.getFullTableName(metadataNamespace, NAMESPACES_TABLE))
+                    .exclusiveStartKey(lastEvaluatedKey)
+                    .build());
+        lastEvaluatedKey = scanResponse.lastEvaluatedKey();
+
+        for (Map<String, AttributeValue> namespace : scanResponse.items()) {
+          String prefixedNamespaceName = namespace.get(NAMESPACES_ATTR_NAME).s();
+          String nonPrefixedNamespaceName =
+              prefixedNamespaceName.substring(namespacePrefix.length());
+          namespaceNames.add(nonPrefixedNamespaceName);
+        }
+      } while (!lastEvaluatedKey.isEmpty());
+
+      return namespaceNames;
+    } catch (ResourceNotFoundException e) {
+      return Collections.emptySet();
+    } catch (Exception e) {
+      throw new ExecutionException("Retrieving the existing namespace names failed", e);
+    }
+  }
+
+  private void createNamespacesTableIfNotExists(boolean noBackup) throws ExecutionException {
+    if (namespacesTableExists()) {
+      return;
+    }
+
+    List<AttributeDefinition> columnsToAttributeDefinitions = new ArrayList<>();
+    columnsToAttributeDefinitions.add(
+        AttributeDefinition.builder()
+            .attributeName(NAMESPACES_ATTR_NAME)
+            .attributeType(ScalarAttributeType.S)
+            .build());
+    try {
+      client.createTable(
+          CreateTableRequest.builder()
+              .attributeDefinitions(columnsToAttributeDefinitions)
+              .keySchema(
+                  KeySchemaElement.builder()
+                      .attributeName(NAMESPACES_ATTR_NAME)
+                      .keyType(KeyType.HASH)
+                      .build())
+              .provisionedThroughput(
+                  ProvisionedThroughput.builder()
+                      .readCapacityUnits(METADATA_TABLES_REQUEST_UNIT)
+                      .writeCapacityUnits(METADATA_TABLES_REQUEST_UNIT)
+                      .build())
+              .tableName(ScalarDbUtils.getFullTableName(metadataNamespace, NAMESPACES_TABLE))
+              .build());
+    } catch (Exception e) {
+      throw new ExecutionException("Creating the namespaces table failed", e);
```

Comment:
How about adding the namespace name in this error message just in case?

Reply:
I will revise this in another PR. 
cf. https://github.com/scalar-labs/scalardb/pull/989#discussion_r1286624632